Server/src/services/DataUpload/EditUploadVerificationService.ts

```diff
@@ -0,0 +1,53 @@
+import { BadRequest } from "@tsed/exceptions";
+import { IDataSetModel } from "../../genericInterfaces/DataProcessInterfaces";
+import { IResponse } from "../../genericInterfaces/ResponsesInterface";
+import { DatasetCommonModel } from "../../models/DatasetModels/DatasetCommonModel";
+import AbstractUploadService from "./AbstractUploadService";
+import EditUploadService from "./EditUploadService";
+
+export class EditUploadVerificationService {
+    private dataService: AbstractUploadService
+    private requestResponse: IResponse
+    private datasetCommonModel: DatasetCommonModel
+
+    constructor() {
+        this.datasetCommonModel = new DatasetCommonModel();
+        this.requestResponse = {} as any
+    }
+
+    async verifyUploader(userId: number, permissionLevel: number, dataset: IDataSetModel, datasetId: number): Promise<IResponse> {
+        try {
+            let clearFlag = false
+            if ((permissionLevel == 1 || permissionLevel == 2) == false) {
+                let verifyUploader = await this.datasetCommonModel.verifyUnapprovedDatasetUploader(datasetId, userId)
+                if (verifyUploader != true) {
+                    throw new BadRequest(verifyUploader)
+                }
+                else {
+                    clearFlag = true
+                }
+            }
+            else {
+                let verifyDatasetExists = await this.datasetCommonModel.verifyDatasetExists(datasetId)
+                if (verifyDatasetExists == undefined || verifyDatasetExists == null) {
+                    throw new BadRequest("No such data set exists")
+                }
+            }
+            this.dataService = new EditUploadService(dataset, datasetId, null)

```

Comment:
This was what I was referring to above. The method can stop here and then call the EditUploadService inside the controller